COOKBOOK.md

```diff
@@ -0,0 +1,964 @@
+
+You can modify `MessageInput` in a large variety of ways, the type definitions for the props give clear insight into all of the options. You can replace the `Input` wholesale, as above, or create you own `MessageInput` component using the provided hooks to access context.
+
+<table>
+  <tr>
+    <td align='center'><img src='./screenshots/cookbook/SendButton.png' width="300"/></td>
+    <td align='center'><img src='./screenshots/cookbook/HasPickers.png' width="300"/></td>
+    <td align='center'><img src='./screenshots/cookbook/NumberOfLines.png' width="300"/></td>
+  </tr>
+  <tr></tr>
+  <tr>
+    <td align='center'>Replace SendButton</td>
+    <td align='center'>hasFilePicker & hasImagePicker - false</td>
+    <td align='center'>numberOfLines={2}</td>
+  </tr>
+</table>
+
+**NOTE:** The `additionalTextInputProps` prop of both `Channel` and `MessageInput` is passed the the internal [`TextInput`](https://reactnative.dev/docs/textinput) component from `react-native`. If you want to change the `TextInput` component props directly this can be done using this prop.
+
+### Disable autocomplete feature on input (mentions and commands)
+
+The auto-complete trigger settings by default include `/`, `@`, and `:` for slash commands, mentions, and emojis respectively. These triggers are created by the exported function `ACITriggerSettings`, which takes `ACITriggerSettingsParams` and returns `TriggerSettings`. You can override this function to remove some or all of the trigger settings via the `autoCompleteTriggerSettings` prop on `Channel`. If you remove the slash commands it is suggested you also remove the commands button using the prop on `Channel` `hasCommands`. You can remove all of the commands by returning an empty object from the function given to `autoCompleteTriggerSettings`.
+
+```tsx
+<Channel
+  channel={channel}
+  keyboardVerticalOffset={headerHeight}
+  thread={thread}
+  hasCommands={false}
+  autoCompleteTriggerSettings={() => ({})}
+>
+```
+
+### Push Notifications
+
+#### Setup
+
+##### iOS
+
+- https://getstream.io/chat/docs/push_ios/?language=java
+- https://getstream.io/chat/docs/rn_push_initial/?language=java
+- https://getstream.io/chat/docs/rn_push_ios/?language=java
+
+##### Android
+
+- https://getstream.io/chat/docs/push_android/?language=java
+- https://getstream.io/chat/docs/rn_push_initial/?language=java
+- https://getstream.io/chat/docs/rn_push_android/?language=java
+
+#### Requirements
+
+- A user must be a member of channel if they expect a push notification for a message on that channel to be sent.
+
+- We only send a push notification when a user is **NOT** connected to chat, i.e. if a user **does NOT have any active WS (websocket) connection**. A WS connection is established when you call `connectUser` on the instance of `StreamChat`. You must also call `addDevice` with the users device token on the instance of `StreamChat` to provide the token to the server for push.
```

Comment:
we establish WS on `getInstance` now I thought?

Reply:
Naw get instance just gives the current or now instance of StreamChat, `this._setupConnection();` is called in connectUser